### DISCUSSION TITLE
How to verify what code is programmed into memory

### DISCUSSION THREAD

**Discussion by Dandy-Don** [micropython/micropython#14500] [RP2040 / Pico]
I'm curious how to know what code is programmed into the RP2040 board.

I sort of understand that CRC can be used on Atmel & Microchip microcontrollers to verify what code has been programmed into the chip.  Is there a way to do something like this using Micropython & RP2040 chips?

**Comment by scruss** [comment]
[picotool verify](https://github.com/raspberrypi/picotool) verifies the uf2 file on your local disk against the contents of the RP2040.

The [USB Flashing Format (UF2)](https://github.com/microsoft/uf2) is rather complicated compared to Atmel chip dumps, and may not be useful for direct binary comparison

**Reply by Dandy-Don** [reply]
Thanks, scruss!

I've never used picotool.  I'll have a look & see how to use it....
